web/ui/dashboard/src/app/private/pages/project/event-logs/event-logs.component.ts

```diff
@@ -181,21 +181,32 @@ export class EventLogsComponent implements OnInit {
 		}
 
 		clearInterval(this.getEventsInterval);
-		if (Object.keys(this.queryParams).length === 0) this.refetchEvents(this.queryParams);
+		this.checkIfTailModeIsEnabled(this.queryParams);
+	}
+
+	checkIfTailModeIsEnabled(data?: FILTER_QUERY_PARAM) {
+		const tailModeConfig = localStorage.getItem('EVENT_LOGS_TAIL_MODE');
+		this.enableTailMode = tailModeConfig ? JSON.parse(tailModeConfig) : false;
+		if (this.enableTailMode) this.getEventLogs({ ...data, showLoader: true }).then(() => this.getEventsAtInterval({ ...data }));
+		else this.getEventLogs({ ...data, showLoader: true });
+	}
+
+	toggleTailMode(e: any) {
+		const tailModeConfig = e.target.checked;
+		this.enableTailMode = tailModeConfig;
+		localStorage.setItem('EVENT_LOGS_TAIL_MODE', JSON.stringify(tailModeConfig));
+		const data = this.getFiltersFromURL();
+		if (tailModeConfig) this.getEventLogs({ ...data, showLoader: true }).then(() => this.getEventsAtInterval({ ...data }));
```

Comment:
This method should not be called here, this function should simply toggle tail mode and return the status to whoever called it.